File: Admin_p/Admin_p.Web/Modules/electroDB/Users/UsersColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace Admin_p.electroDB.Columns;

[ColumnsScript("electroDB.Users")]
[BasedOnRow(typeof(UsersRow), CheckNames = true)]
public class UsersColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int UserId { get; set; }
    [EditLink]
    public string UserName { get; set; }
    public string Email { get; set; }
    public string PasswordHash { get; set; }
    public string Role { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string ShippingAddress { get; set; }
    public string PhoneNumber { get; set; }
}

File: Admin_p/Admin_p.Web/Modules/ServerTypes/electroDB/UsersColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UsersRow } from "./UsersRow";

export interface UsersColumns {
    UserId: Column<UsersRow>;
    UserName: Column<UsersRow>;
    Email: Column<UsersRow>;
    PasswordHash: Column<UsersRow>;
    Role: Column<UsersRow>;
    CreatedAt: Column<UsersRow>;
    UpdatedAt: Column<UsersRow>;
    ShippingAddress: Column<UsersRow>;
    PhoneNumber: Column<UsersRow>;
}

export class UsersColumns extends ColumnsBase<UsersRow> {
    static readonly columnsKey = 'electroDB.Users';
    static readonly Fields = fieldsProxy<UsersColumns>();
}